app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,32 @@
+# responsible for creating and deleting user's favorite tracks
+# depends on asset_id
+# user must be logged_in
+# when favorite track is created it triggers favorites_count counter cache on asset
+class FavoritesController < ApplicationController
+  before_action :find_user
+  before_action :find_asset
+  before_action :require_login
+  before_action :find_favorite_track, only: :destroy
+
+  def create
+    raise ArgumentError unless @asset
+
+    @user.tracks.favorites.create(asset_id: @asset.id)
+    head :ok
+  end
+
+  def destroy
+    @favorite_track.destroy
+    head :ok
+  end
+
+  private
+
+  def find_favorite_track
+    @favorite_track = Track.find_by_id(params[:id])
```

Comment:
We probably want to scope through user here, to make sure user's can't delete any track id.